spec.html

```diff
@@ -25192,9 +25210,9 @@ <h1>Function.prototype.toString ( )</h1>
         <p>When the `toString` method is called, the following steps are taken:</p>
         <emu-alg>
           1. Let _func_ be the *this* value.
-          1. If _func_ is a <emu-xref href="#sec-built-in-function-objects">built-in function object</emu-xref>, return an implementation-defined String source code representation of _func_. The representation must have the syntax of a |NativeFunction|. Additionally, if _func_ has an [[InitialName]] internal slot and _func_.[[InitialName]] is a String, the portion of the returned String that would be matched by |NativeFunctionAccessor?| |PropertyName| must be the value of _func_.[[InitialName]].
```

Comment:
(Coming from https://github.com/tc39/ecma262/issues/2608#issuecomment-1004515458)

I see why this change was necessary: default constructors that are created using `CreateBuiltinFunction` now are "built-in functions" and have an [[InitialName]] slot; however they should still return the [[SourceText]] from the user's program.

But the new spec text does permit implementations to return the [[SourceText]] for basically arbitrary built-in functions now, especially for those that are implemented as ECMAScript function objects. I do not think that change was intentional, I believe that all functions which are not defined by user code (but rather, by the implementation) should have to return a string in `NativeFunction` format.

Would it be sensible to rollback this swapping of `Function.prototype.toString` steps, and instead change `CreateBuiltinFunction` to not create an [[InitialName]] slot for default constructors?

Reply:
Why would implementations be *forced* to omit source text? An engine should always be free to provide meaningful source text in `.toString` if it so chooses.

Reply:
I think it's the (not unreasonable) general expectation that native functions can be detected by looking for `[native code]` in the `toString` representation, and this is used in many codebases. See https://davidwalsh.name/detect-native-function or https://stackoverflow.com/questions/6598945/detect-if-function-is-native-to-browser for reference.
The [*Function.prototype.toString revision* proposal](https://tc39.es/Function-prototype-toString-revision/) did mean to "*standardise the string representation of built-in functions and host objects*". There has gone a lot of thought into this, and there were requests to also ensure proper function name representations and even exact whitespace patterns in the `NativeFunction` syntax.

If you want to change this, fine, but it seems this happened without discussion in TC39, likely even by accident.

Reply:
That is an unreasonable expectation, since `(function () {}).bind()` produces the same toString representation.

To my recollection, it was very much not an accident to _permit_ native functions to have any source text they want, as long as it's either valid ecmascript, or, the native code representation.